apps/server/src/modules/board/controller/api-test/content-element-create.api.spec.ts

```diff
@@ -30,108 +34,190 @@ describe(`content element create (api)`, () => {
+
+				const response = await loggedInClient.post(`${cardNode.id}/elements`, {
+					type: ContentElementType.RICH_TEXT,
+					toPosition: 'not a number',
+				});
 
-			const response = await loggedInClient.post(`${cardNode.id}/elements`, {
-				type: ContentElementType.SUBMISSION_CONTAINER,
+				expect(response.statusCode).toEqual(400);
 			});
 
-			expect((response.body as AnyContentElementResponse).type).toEqual(ContentElementType.SUBMISSION_CONTAINER);
-			expect((response.body as SubmissionContainerElementResponse).content.dueDate).toBeNull();
+			it('should throw an error if toPosition param is a negative number', async () => {
+				const { loggedInClient, cardNode } = await setup();
+
+				const response = await loggedInClient.post(`${cardNode.id}/elements`, {
+					type: ContentElementType.RICH_TEXT,
+					toPosition: -1,
+				});
+
+				expect(response.statusCode).toEqual(400);
+			});
 		});
 
-		it('should actually create the content element', async () => {
-			const { loggedInClient, cardNode } = await setup();
-			const response = await loggedInClient.post(`${cardNode.id}/elements`, { type: ContentElementType.RICH_TEXT });
+		describe('when parent is a submission item', () => {
+			const setup = async () => {
+				await cleanupCollections(em);
 
-			const elementId = (response.body as AnyContentElementResponse).id;
+				const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher();
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
 
-			const result = await em.findOneOrFail(RichTextElementNode, elementId);
-			expect(result.id).toEqual(elementId);
-		});
+				const course = courseFactory.build({ teachers: [teacherUser], students: [studentUser] });
 
-		it('should throw an error if toPosition param is not a number', async () => {
-			const { loggedInClient, cardNode } = await setup();
+				await em.persistAndFlush([teacherAccount, teacherUser, studentAccount, studentUser, course]);
+
+				const columnBoardNode = columnBoardNodeFactory.buildWithId({
+					context: { id: course.id, type: BoardExternalReferenceType.Course },
+				});
+				const columnNode = columnNodeFactory.buildWithId({ parent: columnBoardNode });
+				const cardNode = cardNodeFactory.buildWithId({ parent: columnNode });
+				const submissionElementContainerNode = submissionContainerElementNodeFactory.buildWithId({ parent: cardNode });
+				const submissionItemNode = submissionItemNodeFactory.buildWithId({
+					parent: submissionElementContainerNode,
+					userId: studentUser.id,
+				});
+
+				await em.persistAndFlush([
+					columnBoardNode,
+					columnNode,
+					cardNode,
+					submissionElementContainerNode,
+					submissionItemNode,
+				]);
+				em.clear();
 
-			const response = await loggedInClient.post(`${cardNode.id}/elements`, {
-				type: ContentElementType.RICH_TEXT,
-				toPosition: 'not a number',
+				const loggedInClient = await testApiClientSubmission.login(studentAccount);
```

Comment:
I have added more tests to cover the cases of not authorized user.
But I kept the separation for the parent node (card / submission item). This is because the routes are diferent in these 2 cases, they go to different controllers, and it makes it easier to split it and follow the cases like this.